### PR TITLE
RHMAP-1502 - Provide FHC CLI functionality for UPS

### PR DESCRIPTION
**Pull request by camilamacedo86**
LINK: https://issues.jboss.org/browse/RHMAP-1502

**Commands/Tests**

* Check the UPS commands (fhc help) - `./bin/fhc.js ups `

```
Camila:fh-fhc cmacedo$ ./bin/fhc.js ups 
No action specified. Usage: 
fhc ups <action>
Where <action> is one of:
  config             	Read Push Notifications init config for an app
  read               	Read the Push Notification for an app
  reset              	Reset Push Notifications Secret for an app
  send               	Send Push Notifications for an app
  variant
    add                	List Push Notifications Variants for an app
    delete             	Delete Push Notifications Variant for an app
    list               	List Push Notifications Variants for an app
Camila:fh-fhc cmacedo$ 
```

* UPS init config - `./bin/fhc.js ups config --app=<appId> `

screen shot 2018-01-02 at 5 35 52 pm

* UPS read - `./bin/fhc.js ups read --app=<appId> `

screen shot 2018-01-02 at 5 36 35 pm

* UPS reset - `./bin/fhc.js ups reset--app=<appId> `

```
Camila:fh-fhc cmacedo$ ./bin/fhc.js ups reset --app=xa7cxwzw2sgmjjw7jumkwava
Master Secrete reset/re-newed successfully.
Camila:fh-fhc cmacedo$ 

```

* UPS Variants:
** Add: 
*** Ios: ` ./bin/fhc.js ups variant add --app=xa7cxwzw2sgmjjw7jumkwava --name=test --type=ios --passphrase=test --certificate=test.p12` 

```
Camila:fh-fhc cmacedo$ ./bin/fhc.js ups variant add  --app=rdztoiqowjhmzyl33iijruhm --name=testFHC --type=ios --production=true --certificate=/Users/cmacedo/Desktop/push.p12 --passphrase=redhat
Variant 'testFHC' created successfully
Camila:fh-fhc cmacedo$ 
```

```
Camila:fh-fhc cmacedo$  ./bin/fhc.js ups variant add --app=xa7cxwzw2sgmjjw7jumkwava --name=test --type=ios --passphrase=test --certificate="/Users/cmacedo/Desktop/test.p12"
fhc ERR! Error: Error - not 2xx status code. (400)
fhc ERR! {"certificatePassphraseValid":"the provided certificate passphrase does not match with the uploaded certificate"}
fhc ERR! Request ID:  c4761224-08cb-4298-b477-6891c770c16b
fhc ERR! 
fhc ERR! System Darwin 15.6.0
fhc ERR! command "/Users/cmacedo/.nvm/versions/node/v4.4.7/bin/node" "/Users/cmacedo/work/fh-fhc/bin/fhc.js" "ups" "variant" "add" "--app=xa7cxwzw2sgmjjw7jumkwava" "--name=test" "--type=ios" "--passphrase=test" "--certificate=/Users/cmacedo/Desktop/test.p12"
```

*** Android: ` ./bin/fhc.js ups variant add --app=<appId> --name=<name> --type=android --serverKey=<serverKey> --senderId=<senderId>` 

```
Camila:fh-fhc cmacedo$  ./bin/fhc.js ups variant add --app=xa7cxwzw2sgmjjw7jumkwava --name=test --type=android --serverKey=AIza5a448c2f31700a466fbc9d33d33942b043a27596 --senderId=test@gmail.com
{
  "id": "46957603-207f-4571-b01d-f89511b2753b",
  "name": "test",
  "description": null,
  "variantID": "9be88c97-0d55-42db-b58a-c2c48dd6688d",
  "secret": "e94e2f63-9f0d-4d11-bd0d-574f9b0c9678",
  "developer": "support",
  "googleKey": "AIza5a448c2f31700a466fbc9d33d33942b043a27596",
  "projectNumber": "test@gmail.com",
  "type": "android"
}
Camila:fh-fhc cmacedo$ 
```

** List: `./bin/fhc.js ups variant list --app=<appId>`
screen shot 2018-01-02 at 5 55 44 pm

** Delete: `./bin/fhc.js ups variant delete --app=<appId> id=<VariantID>--type=<android|ios>`
Camila:fh-fhc cmacedo$  ./bin/fhc.js ups variant delete --app=xa7cxwzw2sgmjjw7jumkwava  --id=9be88c97-0d55-42db-b58a-c2c48dd6688d --type=android
Variant deleted successfully.
Camila:fh-fhc cmacedo$ 

*UPS Send Message: `./bin/fhc.js ups send --app=<appId> --message=<message>`

```
Camila:fh-fhc cmacedo$ ./bin/fhc.js ups send --app=rdztoiqowjhmzyl33iijruhm --message="text"
{}
Camila:fh-fhc cmacedo$ 

```

### CODE REVIEW

File: lib/cmd/fh3/ups/config.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    'app' : i18n._("Unique 24 character GUID of your application."),
+    'json' : i18n._("Output in json format")
+  },
+  'url': function(params) {
+    return '/api/v2/ag-push/init/' + params.app;
+  },
+  'method': 'get',
+  'postCmd': function(params, response, cb) {
+    if (!response || !response.pushApplicationID) {
+      return cb(i18n._("Not found Push Notification"));
```
Comment (pb82) [inline lib/cmd/fh3/ups/config.js]: Better: `Push notification not found`

Comment (camilamacedo86) [reply lib/cmd/fh3/ups/config.js]: 👍 Done tks.

**Comment by camilamacedo86** [comment]
@davidffrench thank you. I think here maybe the best ones are the QE team who request it.  @trepel @flashlee could you check these ones?

**Comment by camilamacedo86** [comment]
Hi @damienomurchu, could give a help here? Could you check this one?

**Comment by camilamacedo86** [comment]
Tks @flashlee for the check and review. I will merge it so after a rebase :-)

**Pull request merged**